nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -42,24 +49,26 @@ def system_dir(profile, generation):
 efi /efi/memtest86/BOOTX64.efi
 """
 
-def write_loader_conf(profile, generation):
+def generation_conf_filename(profile, generation, specialisation):
+    profile_part = "-%s" % profile if profile else ""
+    specialisation_part = "-specialisation-%s" % specialisation if specialisation else ""
+    return "nixos{profile_part}{specialisation_part}-generation-{generation}.conf".format(**locals())
```

Comment:
Using python f-strings instead of `.format()` is probably a better choice here.

Reply:
Yeah, certainly. I think in general that's probably the case with all the format strings here, but I'll just clean up my new additions (and this... particularly egregious example)